### ISSUE TITLE
Support instrumenting bindAndHandle for akka-http-server

### ISSUE THREAD

**Issue by kosii** [DataDog/dd-trace-java#1105]
While the java agent support automatic instrumentation of the `bindAndHandleAsync`, and `bindAndHandleSync` methods, it doesn't not support `bindAndHandle`.
Unfortunately the two former are part of the low-level API, but we'd rather stick with the high level akka-http server API. 

There was already one [question](https://github.com/DataDog/dd-trace-java/issues/896#issuecomment-517847174) in a comment for a different issue regarding this, but it would be nice to have a follow up.

**Comment by tylerbenson** [comment]
I have looked at this several times in the past and haven't been able to figure out how to solve it, likely due to my lack of understanding of how Flows work.  Since you are probably more experienced with that, If you have some time, I would appreciate if you submit a PR with a POC on how the instrument at that level should work.

**Comment by jdonnelly-apixio** [comment]
Thanks for posting this issue, it was driving us crazy. Switching to bindAndHandleAsync from bindAndHandle worked for us on akka http 10.1.1.

**Comment by DWvanGeest** [comment]
And for anyone finding this, the way to get your routes working with `bindAndHandleAsync` is to use `Route.asyncHandler` to get the needed `HttpRequest => Future[HttpResponse]`.

**Comment by andyczerwonka** [comment]
I assume that this will work for the news Akka API given `bindAndHandleAsync` is deprecated.
```scala
Http().newServerAt(httpHost, httpPort).bind(Route.toFunction(route))
```

**Comment by tylerbenson** [comment]
Related: #1271 #1105 #2012

**Comment by randomanderson** [comment]
This is resolved by #2050
